packages/core/src/actions.ts

```diff
@@ -808,28 +803,34 @@ export function resolveActions<TContext, TEvent extends EventObject>(
             }
           };
         }
-        return resolvedActionObject;
+        return [resolvedActionObject];
+      }
     }
   }
 
-  function processBlock(block: ActionObject<TContext, TEvent>[]) {
-    let resolvedActions: Array<ActionObject<TContext, TEvent>> = [];
-
-    for (const action of block) {
-      const resolved = handleAction(action);
-      if (resolved) {
-        resolvedActions = resolvedActions.concat(resolved);
-      }
-    }
+  function resolveActionBlocks(
+    _actionBlocks: ActionObject<TContext, TEvent>[]
+  ) {
+    const resolvedActions = flatten(_actionBlocks.map(resolveAction));
 
-    deferredToBlockEnd.forEach((action) => {
-      predictableExec!(action, updatedContext, _event);
-    });
-    deferredToBlockEnd.length = 0;
+    // predictableExec can not be undefined if deferredToBlockEnd has items
+    deferredToBlockEnd.forEach((action) =>
+      predictableExec?.(action, updatedContext, _event)
```

Comment:
previously an exception would have been thrown if predictableExec was undefined.
That can never be the case, but maybe it was intended, in case code changes above would introduce a bug?

Then we should probably change it back. But disliked the 'type casting', thats why i wanted to change it